src/storage/engines/FirebaseStorageEngine.ts

```diff
@@ -376,6 +389,96 @@ export class FirebaseStorageEngine extends StorageEngine {
     return allAnswersPresent;
   }
 
+  async getUserManagementData(key:string) {
+    // Get the user-management collection in Firestore
+    const userManagementCollection = collection(this.firestore, 'user-management');
+    // Grabs all user-management data and returns data based on key
+    const userManagementDocs = await getDocs(userManagementCollection)
+      .then((querySnapshot) => {
+        const docsObject:Record<string, DocumentData> = {};
+        querySnapshot.docs.forEach((queryDoc) => {
+          docsObject[queryDoc.id] = queryDoc.data();
+        });
```

Comment:
This should be a map since you're assigning.

E.g.: `const docsObject = Object.fromEntries(querySnapshot.docs.map((queryDoc) => [queryDoc.id, queryDoc.data()]));`